TileRarities.lua

```diff
@@ -0,0 +1,131 @@
+--[[
+    The data for the different chances and multiplier values can be found on Data:Tile_Chances
+]]--
+local p = {}
+
+function p.main(frame)
+	return p._main(frame.args)
+end
+
+function p._main(args)
+	local json = mw.loadJsonData('Data:Tile_Chances')
```

Comment:
also, i would globally scope your imported json data. put it outside the scope of any of these functions, at the top of the file, and write `local TILE_CHANCES = mw.loadJsonData('Data:Tile_Chances')`

then you can refer to the data in multiple functions, for example the one i suggest making at Comment A